functions/Get-DbaLastBackup.ps1

```diff
@@ -0,0 +1,158 @@
+﻿Function Get-DbaLastBackup
+{
+<#
+.SYNOPSIS
+Get date/time for last known backups
+
+.DESCRIPTION
+Retrieves and compares the date/time for the last known backups, as well as the creation date/time for the database.
+
+
+.PARAMETER SqlServer
+The SQL Server that you're connecting to.
+
+.PARAMETER Credential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER Databases
+Return information for only specific databases
+
+.PARAMETER Exclude
+Return information for all but these specific databases
+
+.PARAMETER Detailed
+Shows detailed information
+
+.NOTES
+Copyright (C) 2016 Klaas Vandenberghe (@powerdbaklaas)
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+
+
+.EXAMPLE
+Get-DbaLastBackup -SqlServer ServerA\sql987
+
+Returns a custom object with Server name, Database name, and the date the last time backups were performed
+
+.EXAMPLE
+Get-DbaLastBackup -SqlServer ServerA\sql987 -Detailed | Out-Gridview
+
+Returns a gridview displaying Server, Database, LastFullBU, TimeSinceFullBU, LastDiffBU, TimeSinceDiffBU, LastLogBU, TimeSinceLastLogBU, RecoveryModel, Status, DatabaseCreated, DaysSinceDbCreated
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[PsCredential]$Credential,
+		[switch]$Detailed
+	)
+
+	DynamicParam {
+		if ($SqlServer) {
+			return Get-ParamSqlDatabases -SqlServer $SqlServer[0] -SqlCredential $Credential
+		}
+	}
+
+	BEGIN
+	{
+		# Convert from RuntimeDefinedParameter object to regular array
+		$databases = $psboundparameters.Databases
+		$exclude = $psboundparameters.Exclude
+	}
+
+	PROCESS
+	{
+		foreach ($servername in $SqlServer)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $servername -SqlCredential $Credential

```

Comment:
Failed connections seem to be unhandled and cause the cmdlet to stop processing when list of servers is piped in from Get-SqlRegisteredServerName.